src/components/settings/DataManagement/ImportDialog.tsx

```diff
@@ -0,0 +1,123 @@
+import type { ReactElement, Dispatch, SetStateAction } from 'react'
+
+import ModalDialog from '@/components/common/ModalDialog'
+import { useAppDispatch } from '@/store'
+import { trackEvent, SETTINGS_EVENTS } from '@/services/analytics'
+import { addedSafesSlice } from '@/store/addedSafesSlice'
+import { addressBookSlice } from '@/store/addressBookSlice'
+import { safeAppsSlice } from '@/store/safeAppsSlice'
+import { settingsSlice } from '@/store/settingsSlice'
+import { FileListCard } from '@/components/settings/DataManagement/FileListCard'
+import { useGlobalImportJsonParser } from '@/components/settings/DataManagement/useGlobalImportFileParser'
+import FileIcon from '@/public/images/settings/data/file.svg'
+import { ImportFileUpload } from '@/components/settings/DataManagement/ImportFileUpload'
+
+import css from './styles.module.css'
+import { showNotification } from '@/store/notificationsSlice'
+
+export const ImportDialog = ({
+  onClose,
+  fileName = '',
+  setFileName,
+  jsonData = '',
+  setJsonData,
+}: {
+  onClose?: () => void
+  fileName: string | undefined
+  setFileName: Dispatch<SetStateAction<string | undefined>>
+  jsonData: string | undefined
+  setJsonData: Dispatch<SetStateAction<string | undefined>>
+}): ReactElement => {
+  const dispatch = useAppDispatch()
+  const { addedSafes, addedSafesCount, addressBook, addressBookEntriesCount, settings, safeApps, error } =
+    useGlobalImportJsonParser(jsonData)
+
+  const isDisabled = (!addedSafes && !addressBook && !settings && !safeApps) || !!error
+
+  const handleClose = () => {
+    setFileName(undefined)
+    setJsonData(undefined)
+    onClose?.()
+  }
+
+  const handleImport = () => {
+    if (addressBook) {
+      dispatch(addressBookSlice.actions.setAddressBook(addressBook))
+      trackEvent({
+        ...SETTINGS_EVENTS.DATA.IMPORT_ADDRESS_BOOK,
+        label: addressBookEntriesCount,
+      })
+    }
+    if (addedSafes) {
+      dispatch(addedSafesSlice.actions.setAddedSafes(addedSafes))
+      trackEvent({
+        ...SETTINGS_EVENTS.DATA.IMPORT_ADDED_SAFES,
+        label: addedSafesCount,
+      })
+    }
+
+    if (settings) {
+      dispatch(settingsSlice.actions.setSettings(settings))
+      trackEvent(SETTINGS_EVENTS.DATA.IMPORT_SETTINGS)
+    }
+
+    if (safeApps) {
+      dispatch(safeAppsSlice.actions.setSafeApps(safeApps))
+      trackEvent(SETTINGS_EVENTS.DATA.IMPORT_SAFE_APPS)
+    }
+
+    dispatch(
+      showNotification({
+        variant: 'success',
+        groupKey: 'global-import-success',
+        message: 'Successfully imported data',
+      }),
+    )
+
+    handleClose()
+  }
+
+  return (
+    <ModalDialog open onClose={handleClose} dialogTitle="Data import" hideChainIndicator>
+      <DialogContent>
+        {!jsonData || !fileName ? (
+          <Box mt={2}>
+            <ImportFileUpload setFileName={setFileName} setJsonData={setJsonData} />
+          </Box>
+        ) : (
```

Comment:
What are the steps to see this component inside the Modal? From my tests so far I only ever get into the other case e.g. when dropping a .json into the field.

Reply:
If you click "Import" from the welcome page, it should open the modal with the dropzone. (See recording in the description.)